scripts/preprocess_spec.py

```diff
@@ -16,24 +16,63 @@ def _to_camel_case(s):
     return ''.join(_title(y) for y in s.split("_"))
 
 
-def iterate_through_operations(spec, func):
+def iterate_through_operations(spec, *funcs):
     for k, v in spec['paths'].iteritems():
         for op in _ops:
             if op in v:
-                func(v[op])
+                for func in funcs:
+                    if func(v[op], v):
+                        del v[op]
+
+
+def _has_property(prop_list, property):
+    for prop in prop_list:
+        if prop["name"] == property:
+            return True
+    return False
 
 
 def process_swagger(infile, outfile):
     with open(infile, 'r') as f:
         spec = json.load(f, object_pairs_hook=OrderedDict)
 
-        def strip_tags_from_operation_id(operation):
+        operation_ids = {}
+
+        def make_operation_id_map(op, _):
+            operation_ids[op['operationId']] = op
+
+        def strip_tags_from_operation_id(operation, _):
             operation_id = operation['operationId']
             for t in operation['tags']:
                 operation_id = operation_id.replace(_to_camel_case(t), '')
             operation['operationId'] = operation_id
 
-        iterate_through_operations(spec, strip_tags_from_operation_id)
+        iterate_through_operations(spec, strip_tags_from_operation_id,
+                                   make_operation_id_map)
+
+        def remove_watch_operations(op, parent):
```

Comment:
(No action required)

Consider avoiding the use of complex inline functions (longer than a few lines).  They are effectively impossible to unit test because they can't be referenced outside of their scope.

Speaking of testing, consider adding tests for this script.